### ISSUE TITLE
[dart2js] Crash for switch-await with patterns

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#51828]
`switch (await ...)` with patterns crashes

```dart
awaitSwitch2() async {
  switch (await bar(3)) {
    case == 1:
      return 1;
      break;
    case == 3:
      return 3;
      break;
    default:
      return -1;
  }
}
```
```
Internal Error: The compiler crashed when compiling this element.
awaitSwitch2() async {
^
The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

* the name and version of your operating system,

* the Dart SDK build number (3.0.0-edge.2eba00fabbd4ef7972e352cf2cddf901ff94c6eb), and

* the entire message you see here (including the full stack trace
  below as well as the source location above).

The compiler crashed: Null check operator used on a null value
#0      HBasicBlock.assignCommonDominator (package:compiler/src/ssa/nodes.dart:1006:36)
#1      HGraph.assignDominators (package:compiler/src/ssa/nodes.dart:378:17)
#2      HGraph.finalize (package:compiler/src/ssa/nodes.dart:364:5)
#3      KernelSsaGraphBuilder._closeFunction (package:compiler/src/ssa/builder.dart:1816:11)
#4      KernelSsaGraphBuilder._buildGenerator (package:compiler/src/ssa/builder.dart:1413:9)
#5      KernelSsaGraphBuilder._buildFunctionNode (package:compiler/src/ssa/builder.dart:1322:7)
#6      KernelSsaGraphBuilder.build.<anonymous closure> (package:compiler/src/ssa/builder.dart:468:15)
#7      DiagnosticReporter.withCurrentElement (package:compiler/src/diagnostics/diagnostic_listener.dart:142:15)
#8      KernelSsaGraphBuilder.build (package:compiler/src/ssa/builder.dart:453:21)
...
```

with assertions enabled it appears that we are not tracking some value as a 'variable':
```
The compiler crashed: 'package:compiler/src/ssa/locals_handler.dart': Failed assertion: line 613 pos 12: 'localsHandlers.length > 0': is not true.
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      LocalsHandler.mergeMultiple (package:compiler/src/ssa/locals_handler.dart:613:12)
#3      KernelSsaGraphBuilder.visitLabeledStatement (package:compiler/src/ssa/builder.dart:2794:34)
#4      LabeledStatement.accept (package:kernel/ast.dart:9981:43)
#5      KernelSsaGraphBuilder.visitBlock (package:compiler/src/ssa/builder.dart:1879:17)
#6      Block.accept (package:kernel/ast.dart:9771:43)
#7      KernelSsaGraphBuilder._buildGenerator (package:compiler/src/ssa/builder.dart:1412:28)
#8      KernelSsaGraphBuilder._buildFunctionNode (package:compiler/src/ssa/builder.dart:1322:7)
#9      KernelSsaGraphBuilder.build.<anonymous closure> (package:compiler/src/ssa/builder.dart:468:15)
#10     DiagnosticReporter.withCurrentElement (package:compiler/src/diagnostics/diagnostic_listener.dart:142:15)
#11     KernelSsaGraphBuilder.build (package:compiler/src/ssa/builder.dart:453:21)
...
```

<details><summary>Repro</summary>
<p>

```dart
import 'package:expect/expect.dart';
import "package:async_helper/async_helper.dart";

import 'dart:async';

// It does not matter where a future is generated.
bar(p) async => p;
baz(p) => new Future(() => p);

awaitSwitch1() async {
  switch (await bar(3)) {
    case 1:
      return 1;
      break;
    case 3:
      return 3;
      break;
    default:
      return -1;
  }
}

awaitSwitch2() async {
  switch (await bar(3)) {
    case == 1:
      return 1;
      break;
    case == 3:
      return 3;
      break;
    default:
      return -1;
  }
}

test() async {
  var result;
  for (int i = 0; i < 10; i++) {
    result = await awaitSwitch1();
    Expect.equals(3, result);
    result = await awaitSwitch2();
    Expect.equals(3, result);
  }
}

main() {
  asyncStart();
  test().then((_) {
    asyncEnd();
  });
}
```

</p>
</details>

**Comment by rakudrama** [comment]
Simplified repro: we see that has nothing to do with switch-await except that the CFE lowers that to a labelled statement that has `break` statements in a dead-code position:

```dart
int foo2() {
  var x = 3;
  L:
  {
    if (x == 1) {
      return 1;
      break L;
    }
    if (x == 3) {
      return 2;
      break L;
    }
    {
      return -1;
    }
  }
}

main() {
  final result = foo2();
  if (result != 3) throw 'Bad result: $result';
}
```

**Comment by rakudrama** [comment]
Fixed by 71d6292eb30a7e8b19284231f01dd2dc560c283d
